hypervisor/dm/vpci/vpci.c

```diff
@@ -370,30 +370,101 @@ static void vpci_deinit_pt_dev(struct pci_vdev *vdev)
 	deinit_vmsi(vdev);
 }
 
-static int32_t write_pt_dev_cfg(struct pci_vdev *vdev, uint32_t offset,
-		uint32_t bytes, uint32_t val)
+struct cfg_header_perm {
+	/* per bit for per 4 Bytes registers, only low 16 bit is used.
+	 * for example, bit 0 for CFG Vendor ID and Device ID register,
+	 * Bit 1 for CFG register Command and Status register and so on
+	 */
+
+	/* When one bit set, it means its corresponding 4 Bytes registers is pass through,
+	 * Otherwise, it's virtualized.
+	 */
```

Comment:
check with Shiqing for the correctness of comments.

Reply:
I will ask Shiqing for help.

Reply:
Here is my understanding. Not quite sure if I understand it correctly. 

For each 4-byte register defined in PCI config space header, there is one bit dedicated for it in pt_mask and ro_mask.
For example, bit 0 for CFG Vendor ID and Device ID register, Bit 1 for CFG register Command and Status register, and so on.

For each mask, only low 16-bits takes effect.

If bit x is set the pt_mask, it indicates that the corresponding 4 Bytes register for bit x is pass through to guest. Otherwise, it's virtualized.

Reply:
Thanks shiqing for clear this comments. You're right. I will update the comments. Thanks again.